src/Resources/contao/controllers/BackendIndex.php

```diff
@@ -68,10 +76,30 @@ public function run()
 			$arrParams['referer'] = $referer;
 		}
 
-		$router = \System::getContainer()->get('router');
+		$router = $container->get('router');
 
 		/** @var BackendTemplate|object $objTemplate */
 		$objTemplate = new \BackendTemplate('be_login');
+		$objTemplate->action = ampersand(\Environment::get('request'));
+		$objTemplate->headline = $GLOBALS['TL_LANG']['MSC']['loginBT'];
+
+		/** @var Request $request */
+		$request = $container->get('request_stack')->getCurrentRequest();
+
+		if ($request->getRequestUri() === $router->generate('contao_backend_2fa'))
```

Comment:
This seems wrong to me. The `contao_backend_2fa` route must not be publicly callable by just adding `/contao/2fa` to the URL. The same is true for the `contao_backend_2fa_check` route.

Reply:
```yaml
#security.yml
    access_control:
        - { path: ^/contao/2fa, role: IS_AUTHENTICATED_2FA_IN_PROGRESS }
```

Reply:
See here: 
https://github.com/bytehead/core-bundle/blob/b3109397897e918c40aa48b57e9ec4fa15125a07/src/Controller/BackendController.php#L251-L261

Reply:
Ah, true.